src/enums.rs

```diff
@@ -399,6 +399,64 @@ pub enum Icon {
     },
 }
 
+impl Serialize for Icon {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        #[derive(Serialize, Default)]
+        struct IconObject {
+            #[serde(rename = "type")]
+            pub icon_type: String,
+            #[serde(skip_serializing_if = "Option::is_none")]
+            pub name: Option<String>,
+            #[serde(skip_serializing_if = "Option::is_none")]
+            pub path: Option<PathBuf>,
+            #[serde(skip_serializing_if = "Option::is_none")]
+            pub url: Option<Url>,
+            #[serde(skip_serializing_if = "Option::is_none")]
+            pub width: Option<u32>,
+            #[serde(skip_serializing_if = "Option::is_none")]
+            pub height: Option<u32>,
+        }
```

Comment:
Do we really need this? It was just a hackish thing I did for deserialize just to get the first release rolling; I will remove them from the deserialize part as well

Reply:
Feel free to make a suggestion how to improve this. IMO it's more elegant as doing `s.serialize_field("height", &height)?` manually for every field.

Reply:
The thing is that for other structs/enums, you have done it the other so it makes things looks inconsistent :/